### PR TITLE
Remove transaction execution on transaction pool add

### PR DESCRIPTION
**Pull request by diega**
Please, dear bot, run our *fed:txpool_improvements* integration tests

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/core/bc/AccountInformationProvider.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+public interface AccountInformationProvider {
+
+    /**
+     * Retrieve balance of an account
+     *
+     * @param account of the account
+     * @return balance of the account as a <code>BigInteger</code> value
+     */
+    Coin getBalance(RskAddress account);
```
Comment (lsebrie) [inline rskj-core/src/main/java/co/rsk/core/bc/AccountInformationProvider.java]: I would use address as a var name for this methods on interface

File: rskj-core/src/main/java/co/rsk/core/bc/PendingState.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+
+    public PendingState(Repository repository, TransactionSet pendingTransactions, TransactionExecutorFactory transactionExecutorFactory) {
+        this.pendingRepository = repository.startTracking();
+        this.pendingTransactions = pendingTransactions;
+        this.transactionExecutorFactory = transactionExecutorFactory;
+    }
+
+    @Override
+    public Coin getBalance(RskAddress account) {
```
Comment (lsebrie) [inline rskj-core/src/main/java/co/rsk/core/bc/PendingState.java]: Same here for var names on the implementation

File: rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java [hunk 3]
```diff
@@ -422,10 +423,10 @@ public String eth_getTransactionCount(String address, String blockId) throws Exc
         try {
             RskAddress addr = new RskAddress(address);
 
-            Repository repository = getRepoByJsonBlockId(blockId);
+            AccountInformationProvider addressInformationProvider = getAccountInformationProvider(blockId);
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java]: rename var :slightly_smiling_face:

File: rskj-core/src/test/java/co/rsk/core/bc/TransactionPoolImplTest.java [hunk 4]
```diff
@@ -223,23 +215,8 @@ public void addAndExecutePendingTransaction() {
 
         transactionPool.addTransaction(tx);
 
-        Repository repository = transactionPool.getRepository();
-        Assert.assertEquals(BigInteger.valueOf(1001000), repository.getBalance(receiver.getAddress()).asBigInteger());
-    }
-
-    @Test
```
Comment (lsebrie) [inline rskj-core/src/test/java/co/rsk/core/bc/TransactionPoolImplTest.java]: I would try to keep this test reworked

Comment (diega) [reply rskj-core/src/test/java/co/rsk/core/bc/TransactionPoolImplTest.java]: This very same test is implemented bellow in the `updateTransactionPool` method

**Comment by rskops** [comment]
SonarQube analysis reported 2 issues

1. ![MAJOR][MAJOR] [TransactionPoolImpl.java#L160](https://github.com/rsksmart/rskj/blob/b42bab145e2958703179e66d1bad0b3ba54af29b/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L160): Inconsistent synchronization of co.rsk.core.bc.TransactionPoolImpl.bestBlock; locked 53% of time [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)
1. ![MINOR][MINOR] [TransactionPoolImpl.java#L159](https://github.com/rsksmart/rskj/blob/b42bab145e2958703179e66d1bad0b3ba54af29b/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L159): Rename "repository" which hides the field declared at line 64. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 2 issues

1. ![MAJOR][MAJOR] [TransactionPoolImpl.java#L160](https://github.com/rsksmart/rskj/blob/0a3027fbe02f9375a2a9a7e56e7077387e75867b/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L160): Inconsistent synchronization of co.rsk.core.bc.TransactionPoolImpl.bestBlock; locked 53% of time [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)
1. ![MINOR][MINOR] [TransactionPoolImpl.java#L159](https://github.com/rsksmart/rskj/blob/0a3027fbe02f9375a2a9a7e56e7077387e75867b/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L159): Rename "repository" which hides the field declared at line 64. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 2 issues

1. ![MAJOR][MAJOR] [TransactionPoolImpl.java#L160](https://github.com/rsksmart/rskj/blob/b04dc87d78b17aae444c448e41c5e95b1fdc220d/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L160): Inconsistent synchronization of co.rsk.core.bc.TransactionPoolImpl.bestBlock; locked 53% of time [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AIS2_INCONSISTENT_SYNC)
1. ![MINOR][MINOR] [TransactionPoolImpl.java#L159](https://github.com/rsksmart/rskj/blob/b04dc87d78b17aae444c448e41c5e95b1fdc220d/rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java#L159): Rename "repository" which hides the field declared at line 64. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**